src/utils/generic.rs

```diff
@@ -21,43 +24,56 @@ pub fn nonce(seed_nonce: &Scalar, label: &str, index: Option<usize>) -> Result<S
     let encoded_label = label.as_bytes();
     if encoded_label.len() > 16 {
         // See https://www.blake2.net/blake2.pdf section 2.8
-        return Err(ProofError::InvalidLength("nonce label".to_string()));
-    };
-    let hasher = if let Some(salt) = index {
+        Err(ProofError::InvalidLength("nonce label".to_string()))
+    } else if let Some(salt) = index {
         let encoded_index = salt.to_le_bytes();
         if encoded_index.len() > 16 {
             // See https://www.blake2.net/blake2.pdf section 2.8
-            return Err(ProofError::InvalidLength("nonce index".to_string()));
-        };
-        Blake2b::with_params(&seed_nonce.to_bytes(), &encoded_index, encoded_label)
+            Err(ProofError::InvalidLength("nonce index".to_string()))
+        } else {
+            Ok(Scalar::from_hasher_blake2b(Blake2b::with_params(
+                &seed_nonce.to_bytes(),
+                &encoded_index,
+                encoded_label,
+            )))
+        }
     } else {
-        Blake2b::with_params(&seed_nonce.to_bytes(), &[], encoded_label)
-    };
-
-    Ok(Scalar::from_hasher_blake2b(hasher))
+        Ok(Scalar::from_hasher_blake2b(Blake2b::with_params(
+            &seed_nonce.to_bytes(),
+            &[],
+            encoded_label,
+        )))
+    }
 }
 
 /// Decompose a given value into a vector of scalars for the required bit length
 pub fn bit_vector_of_scalars(value: u64, bit_length: usize) -> Result<Vec<Scalar>, ProofError> {
     if !bit_length.is_power_of_two() || bit_length > RangeProof::MAX_BIT_LENGTH {
-        return Err(ProofError::InvalidLength(
+        Err(ProofError::InvalidLength(
```

Comment:
Same here, nested code reads and follows more difficult than non nested code.

Reply:
See above

Reply:
Reverted